--- a/src/liboslexec/CMakeLists.txt
+++ b/src/liboslexec/CMakeLists.txt
@@ -88,15 +88,11 @@
     # LLVM/libc++ installed -- for some reason, LLVM 3.4 wasn't finding it,
     # so in that specific case, append another -I to point it in the right
     # direction.
-    #if (APPLE AND ${LLVM_BC_GENERATOR} MATCHES ".*clang.*")
-    #    exec_program ( "${LLVM_BC_GENERATOR}" ARGS --version OUTPUT_VARIABLE MY_CLANG_VERSION )
-    #    string (REGEX REPLACE "clang version ([0-9][.][0-9]+).*" "\\1" MY_CLANG_VERSION "${MY_CLANG_VERSION}")
-    #    if ((${MY_CLANG_VERSION} VERSION_GREATER "3.3")
-    #          AND (EXISTS "/usr/lib/libc++.dylib")
-    #          AND (EXISTS "/Library/Developer/CommandLineTools/usr/lib/c++/v1"))
-    #        set (LLVM_COMPILE_FLAGS ${LLVM_COMPILE_FLAGS} "-I/Library/Developer/CommandLineTools/usr/lib/c++/v1")
-    #    endif ()
-    #endif ()
+    if (APPLE)
+      if ((EXISTS "/Library/Developer/CommandLineTools/usr/include/c++/v1"))
+        set (LLVM_COMPILE_FLAGS ${LLVM_COMPILE_FLAGS} "-I/Library/Developer/CommandLineTools/usr/include/c++/v1")
+      endif ()
+    endif ()

     list (TRANSFORM IMATH_INCLUDES PREPEND -I
           OUTPUT_VARIABLE ALL_IMATH_INCLUDES)
