diff --git a/setup.py b/setup.py
index 0b56654..b924d28 100644
--- a/setup.py
+++ b/setup.py
@@ -871,11 +871,6 @@ class pyside_build(_build):
         if run_process(cmd_make) != 0:
             raise DistutilsSetupError("Error compiling " + extension)

-        if extension.lower() == "shiboken2":
-            log.info("Generating Shiboken documentation %s..." % extension)
-            if run_process([self.make_path, "doc"]) != 0:
-                raise DistutilsSetupError("Error generating documentation " + extension)
-
         if not OPTION_SKIP_MAKE_INSTALL:
             log.info("Installing module %s..." % extension)
             # Need to wait a second, so installed file timestamps are older than build file
diff --git a/sources/pyside2/CMakeLists.txt b/sources/pyside2/CMakeLists.txt
index 2326d2b..1e551dd 100644
--- a/sources/pyside2/CMakeLists.txt
+++ b/sources/pyside2/CMakeLists.txt
@@ -385,9 +385,3 @@ if (BUILD_TESTS)
     add_subdirectory(tests)
 endif ()

-find_program(DOT_EXEC dot)
-if (QT_SRC_DIR AND DOT_EXEC)
-    add_subdirectory(doc)
-else ()
-    message(STATUS "QT_SRC_DIR variable not set or graphviz not found, apidoc generation targets disabled.")
-endif()
diff --git a/sources/shiboken2/CMakeLists.txt b/sources/shiboken2/CMakeLists.txt
index 96d4ec5..3570c3f 100644
--- a/sources/shiboken2/CMakeLists.txt
+++ b/sources/shiboken2/CMakeLists.txt
@@ -213,7 +213,6 @@ else()
 endif()

 add_subdirectory(libshiboken)
-add_subdirectory(doc)

 # deps found, compile the generator.
 if (Qt5Core_FOUND AND PYTHONINTERP_FOUND)
