--- a/src/appleseed/foundation/platform/path.h	2019-08-31 16:49:01.000000000 +0100
+++ b/src/appleseed/foundation/platform/path.h	2022-01-25 18:01:04.042637656 +0000
@@ -40,6 +40,7 @@
 #include "main/dllsymbol.h"
 
 // Boost headers.
+#include "boost/filesystem/exception.hpp"
 #include "boost/filesystem/operations.hpp"
 #include "boost/filesystem/path.hpp"
 
--- a/src/appleseed/foundation/utility/benchmark/benchmarkaggregator.cpp	2019-08-31 16:49:01.000000000 +0100
+++ b/src/appleseed/foundation/utility/benchmark/benchmarkaggregator.cpp	2022-01-25 18:03:50.692067431 +0000
@@ -40,6 +40,7 @@
 
 // Boost headers.
 #include "boost/date_time/posix_time/posix_time.hpp"
+#include "boost/filesystem/directory.hpp"
 #include "boost/filesystem/operations.hpp"
 #include "boost/filesystem/path.hpp"
 #include "boost/regex.hpp"

--- a/src/appleseed/renderer/modeling/project/assethandler.cpp	2019-08-31 16:49:01.000000000 +0100
+++ b/src/appleseed/renderer/modeling/project/assethandler.cpp	2022-01-25 18:07:51.654134776 +0000
@@ -80,7 +80,7 @@
         try
         {
             create_directories(dest_path.parent_path());
-            copy_file(source_path, dest_path, copy_option::overwrite_if_exists);
+            copy_file(source_path, dest_path, copy_options::overwrite_existing);
             return true;
         }
         catch (const std::exception& e)     // namespace qualification required

--- a/src/appleseed/renderer/modeling/project/projectfilewriter.cpp	2019-08-31 08:49:01.000000000 -0700
+++ b/src/appleseed/renderer/modeling/project/projectfilewriter.cpp	2022-05-01 09:47:34.000000000 -0700
@@ -125,7 +125,7 @@
             const char*         filepath,
             FILE*               file,
             const int           options)
-          : m_project_new_root_dir(filesystem::path(filepath).parent_path())
+          : m_project_new_root_dir(bf::path(filepath).parent_path())
           , m_file(file)
           , m_options(options)
           , m_indenter(4)
@@ -152,7 +152,7 @@
         }
 
       private:
-        const filesystem::path  m_project_new_root_dir;
+        const bf::path  m_project_new_root_dir;
         FILE*                   m_file;
         const int               m_options;
         Indenter                m_indenter;
