--- a/src/appleseed/renderer/kernel/rendering/final/texturecontrolledpixelrenderer.cpp	2022-02-03 14:23:35.584430758 +0000
+++ b/src/appleseed/renderer/kernel/rendering/final/texturecontrolledpixelrenderer.cpp	2022-02-03 14:25:26.098466723 +0000
@@ -61,6 +61,7 @@
 // OpenImageIO headers.
 #include "foundation/platform/_beginoiioheaders.h"
 #include "OpenImageIO/imagebufalgo.h"
+#include "OpenImageIO/imagecache.h"
 #include "OpenImageIO/typedesc.h"
 #include "foundation/platform/_endoiioheaders.h"

--- a/cmake/modules/FindOpenImageIO.cmake	2022-03-03 10:25:16.000000000 +0000
+++ b/cmake/modules/FindOpenImageIO.cmake	2022-03-03 10:24:35.000000000 +0000
@@ -57,7 +57,7 @@
 # Set the output variables.
 if (OPENIMAGEIO_FOUND)
     set (OPENIMAGEIO_INCLUDE_DIRS ${OPENIMAGEIO_INCLUDE_DIR})
-    set (OPENIMAGEIO_LIBRARIES ${OPENIMAGEIO_LIBRARY})
+    set (OPENIMAGEIO_LIBRARIES ${OPENIMAGEIO_LIBRARY} OpenImageIO_Util)
 else ()
     set (OPENIMAGEIO_INCLUDE_DIRS)
     set (OPENIMAGEIO_LIBRARIES)
