diff --git a/src/IECoreGL/Shader.cpp b/src/IECoreGL/Shader.cpp
index 207c12077..bc2d3f7ca 100644
--- a/src/IECoreGL/Shader.cpp
+++ b/src/IECoreGL/Shader.cpp
@@ -181,7 +181,7 @@ class Shader::Implementation : public IECore::RefCounted
 						}
 					}
 
-					if( p.type == GL_SAMPLER_2D )
+					if( p.type == GL_SAMPLER_2D || p.type == GL_SAMPLER_3D )
 					{
 						// we assign a specific texture unit to each individual
 						// sampler parameter - this makes it much easier to save

