--- a/build/macos.inc	2020-07-10 06:18:01.000000000 -0700
+++ b/build/macos.inc	2021-12-04 11:17:43.000000000 -0800
@@ -36,11 +36,15 @@
      export arch:=ppc32
    endif
  else
+  ifeq ($(shell /usr/sbin/sysctl -n hw.machine),arm64)
+   export arch:=arm64
+  else
    ifeq ($(shell /usr/sbin/sysctl -n hw.optional.x86_64 2>/dev/null),1)
      export arch:=intel64
    else
      export arch:=ia32
    endif
+  endif
  endif
 endif