--- a/src/scene/object.h 2022-05-07 14:59:24.033859832 +0100
+++ b/src/scene/object.h 2022-05-07 15:00:03.158409903 +0100
@@ -22,6 +22,7 @@
 /* included as Object::set_particle_system defined through NODE_SOCKET_API does
  * not select the right Node::set overload as it does not know that ParticleSystem
  * is a Node */
+#include "scene/geometry.h"
 #include "scene/particles.h"
 #include "scene/scene.h"
