--- a/source/MaterialXRender/External/OpenImageIO/FindOpenImageIO.cmake	2023-04-21 00:10:52.000000000 +0100
+++ b/source/MaterialXRender/External/OpenImageIO/FindOpenImageIO.cmake	2023-06-05 11:34:36.297457384 +0100
@@ -47,0 +48,7 @@
+
+find_library ( OPENIMAGEIO_UTIL_LIBRARY
+               NAMES OpenImageIO_Util${OIIO_LIBNAME_SUFFIX}
+               HINTS ${OPENIMAGEIO_ROOT_DIR}/lib
+               PATH_SUFFIXES lib64 lib
+               PATHS "${OPENIMAGEIO_ROOT_DIR}/lib" )
+               
@@ -69 +76 @@
-set ( OPENIMAGEIO_LIBRARIES ${OPENIMAGEIO_LIBRARY})
+set ( OPENIMAGEIO_LIBRARIES ${OPENIMAGEIO_LIBRARY} ${OPENIMAGEIO_UTIL_LIBRARY})
